### PR TITLE
Fix OpenAPI specification inconsistencies

### PR DESCRIPTION
**Pull request by mrekucci**
Several inconsistencies declared in the pins and stamps endpoints
of the OpenAPI specification and the bee API server implementation
are removed. On top of that also fixes several declaration errors.

### CODE REVIEW

**Review by anatollupacescu — APPROVED** [review]

**Comment by AuHau** [comment]
👍

**Review by aloknerurkar — APPROVED** [review]

File: openapi/Swarm.yaml [hunk 1]
```diff
@@ -443,7 +443,7 @@ paths:
       - in: path
         name: address
         schema:
-          $ref: "SwarmCommon.yaml#/components/schemas/SwarmAddress"
+          $ref: "SwarmCommon.yaml#/components/schemas/SwarmOnlyReference"
```
Comment (acud) [inline openapi/Swarm.yaml:446]: not sure how this will play out with unpinning an encrypted entry

File: openapi/Swarm.yaml [hunk 2]
```diff
@@ -723,6 +727,13 @@ paths:
           description: Default response
 
   "/stamps/{id}":
+    parameters:
+      - in: path
+        name: id
+        schema:
+          $ref: "SwarmCommon.yaml#/components/schemas/SwarmAddress"
+        required: true
+        description: Swarm address of the stamp
```
Comment (acud) [inline openapi/Swarm.yaml:736]: actually the batchID is not an address, it is just a uuid we get from the chain AFAIK. not sure exactly how it is calculated. @ralph-pichler ?

Comment (ralph-pichler) [reply openapi/Swarm.yaml:736]: it is a 32 byte hash based on the creator address and random nonce

Comment (acud) [reply openapi/Swarm.yaml:738]: so maybe it would be good to have this as a 32 byte hex string (I think we have this sort type in the OpenAPI spec) :+1:

Comment (AuHau) [reply openapi/Swarm.yaml:738]: `SwarmAddress` is just that ;-)

Comment (mrekucci) [reply openapi/Swarm.yaml:738]: Created separate type for that

**Review by acud — APPROVED** [review]

**Pull request merged**